source-map.bs

```diff
@@ -274,6 +290,44 @@ fields represent mapped code that also has a mapped name.
 Note: Using file offsets was considered but rejected in favor of using line/column data to avoid becoming
 misaligned with the original due to platform-specific line endings. 
 
+### [=names=] for generated JavaScript code
+
+Source map generators should create a [=mapping=] entry with a [=name=] field for
+the following JavaScript tokens, if
+
+1. The original source language construct maps semantically to the generated
+    JavaScript code.
+
+1. The original source language construct has a name.
+
+Then the [=name=] of the [=mapping=] entry should be the name of the original
+source language construct.
+
+Example: A TypeScript arrow function compiled to a JavaScript function, or a
+local variable that was renamed.
+
+1. The opening parenthesis `(` of [=FunctionDeclaration=], [=FunctionExpression=],
+    [=AsyncFunctionDeclaration=] and [=AsyncFunctionExpression=] for (async) functions.
+
+1. The opening parenthesis `(` or the first letter of the *BindingExpression*
+    of [=ArrowParameters=] and [=AsyncArrowFunction=] for (async) arrow functions.
+
+1. The opening parenthesis `(` of [=MethodDefinition=] for class methods. For the
+    [=MethodDefinition=] where *PropName* is "constructor", the [=name=] should be
+    the original class name.
+
+1. The opening parenthesis `(` of [=GeneratorDeclaration=], [=GeneratorExpression=],
+    [=AsyncGeneratorDeclaration=] and [=AsyncGeneratorExpression=] for (async) generator functions.
+
+1. The first letter of [=Identifier=] for variables. This includes (but is not limited to)
+    [=LexicalDeclaration=], [=VariableStatement=] and arguments in [=Parameter List=].
```

Comment:
Is "function declaration" in this list?

e.g.

```js
var foo = function () {}
```
```js
function foo() {}
```

to they both have two name mappings? (one for the variable they introduce, and one for the actual function)

Reply:
Isn't the first example a `VariableStatement` plus `FunctionExpression`?

For the second case, yes, we could recommend two name mappings, one on the identifier and one on the paren.